src/contexts/TodoContext.js

```diff
@@ -0,0 +1,66 @@
+import React, { createContext, useState } from 'react'
+
+const TodoContext = createContext()
+
+const TodoProvider = ({ children }) => {
+  const [todos, setTodos] = useState([])
+  const [todo, setTodo] = useState('')
+  const [statusTodo, setStatusTodo] = useState('all')
+
+  const addTodoKeyDown = (event) => {
+    if (event.keyCode === 13 && event.target.value !== '') {
+      addTodo()
+      setTodo('')
+    }
+  }
+
+  const addTodo = () => {
+    const newTodo = { title: todo, status: 'pending' }
+    setTodos([...todos, newTodo])
+  }
+
+  const addTodoButtonClick = () => {
+    if (todo !== '') {
+      addTodo()
+      setTodo('')
+    }
+  }
+
+  const setTodoChange = (event) => setTodo(event.target.value)
+
+  const deleteIconClick = (index) => {
+    const copyTodos = todos.filter((todo) => todos.indexOf(todo) !== index)
+    setTodos([...copyTodos])
+  }
+
+  const todoStatusChange = (index) => {
+    const newTodo = todos.find((todo) => todos.indexOf(todo) === index)
+    const newStatus = newTodo.status === 'pending' ? 'completed' : 'pending'
+    const updateNewTodo = { ...newTodo, status: newStatus }
+    const newTodos = todos.map((todo) => (todos.indexOf(todo) === index ? updateNewTodo : todo))
+    setTodos([...newTodos])
+  }
+
+  return (
+    <TodoContext.Provider
+      value={{
+        todo,
+        todos,
+        setTodo,
+        setTodos,
```

Comment:
`setTodos` no la estás utilizando en ningún lado. Por lo general, cuando tenés un estado con datos complejos (objeto, u array de objetos), lo mejor es no exponer directamente la función que te permite modificar ese estado, sino exponer funciones que modifican ese estado indirectamente. De esta forma, sólo permitís las operaciones que querés, y tenés más control sobre ellas. En este caso, ya tenés las operaciones que te permiten hacerlo: `addTodo`, `deleteTodo`, `changeTodoStatus`.

```suggestion
        addTodo,
        deleteTodo,
        changeTodoStatus,
```